concordium-consensus/src/Concordium/GlobalState/Basic/TreeState.hs

```diff
@@ -293,7 +293,13 @@ instance (bs ~ BlockState m, BS.BlockStateStorage m, Monad m, MonadIO m, MonadSt
           Just (_, Finalized{}) -> return TS.ObsoleteNonce
           Just (tr', results) -> do
             when (slot > results ^. tsSlot) $ transactionTable . ttHashMap . at' trHash . mapped . _2 %= updateSlot slot
-            return $ TS.Duplicate tr'
+            return $ TS.Duplicate tr' $ getVerRes results
+      where
+        getVerRes :: TransactionStatus -> Maybe TVer.VerificationResult
+        getVerRes status = case status of
+          Received _ verRes -> Just verRes
+          Committed _ verRes _ -> Just verRes
+          Finalized {} -> Nothing
```

Comment:
I think it would be better to just inline this. It is used once and its meaning would be clearer.

Alternatively document it a bit so the strange semantics is clarified.